Dockerfile

```
@@ -29,15 +30,22 @@ RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build \
 FROM alpine:latest
 
 # Install runtime dependencies
+# - docker-cli: For core component to manage containers
+# - pass, gnupg: For secrets component (password-store backend)
+# - ca-certificates: TLS support
+# - curl, wget: Health checks and debugging
+# - tzdata: Timezone support
 RUN apk add --no-cache \
```

Hadolint (2.14.0)

[warning] Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`

(DL3018)